elcid/assets/js/elcid/services/exception_handler.js

```diff
@@ -0,0 +1,13 @@
+angular.module('opal.services').factory('$exceptionHandler', function($injector, $log) {
+    "use strict";
+    var errorCatcherHandler = function errorCatcherHandler(exception, cause) {
+        // we use the injector to stop a circular dependency as per
+        // https://stackoverflow.com/questions/22332130/injecting-http-into-angular-factoryexceptionhandler-results-in-a-circular-de
+        var $http = $injector.get("$http");
+        var url = "/elcid/v0.1/error_emailer/"
+        $http.post(url, {exception: exception, cause: cause});
```

Comment:
Do we have an illustrative value of the `cause` argument ?

Reply:
```
<div class="outer-container ng-scope" ng-view="" autoscroll="true" ng-hide="ready==false">
```

ie it gives you the element from where you're declaring the scope that threw the error. Not really that useful for us, but I figured more data is better than less. If we ever move to having multiple angular apps it might be useful it could be handy.